spyder/widgets/tests/helpers.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+
+"""
+Helpers used in widget tests
+"""
+
+
+SHORTCUTS = {
+    'Close pane': ('Ctrl+Shift+F4' , u'⇧⌘F4'),
+    'Configure': ('Ctrl+F6', u'⌘F6'),
+    'Debug': ('Ctrl+F5', u'⌘F5'),
+    'Debug continue': ('Ctrl+F12', u'⌘F12'),
+    'Debug exit': ('Ctrl+Shift+F12', u'⇧⌘F12'),
+    'Debug step into': ('Ctrl+F11', u'⌘F11'),
+    'Debug step over': ('Ctrl+F10', u'⌘F10'),
+    'Debug step return': ('Ctrl+Shift+F11', u'⇧⌘F11'),
+    'File switcher': ('Ctrl+P', u'⌘P'),
+    'Find next': ('F3', u'F3'),
+    'Find previous': ('Shift+F3', u'⇧F3'),
+    'Find text': ('Ctrl+F', u'⌘F'),
+    'Fullscreen mode': ('F11', u'F11'),
+    'Hide find and replace': ('Escape', u'⎋'),
+    'Layout preferences': ('Shift+Alt+P', u'⌥⇧P'),
+    'Lock unlock panes': ('Ctrl+Shift+F5', u'⇧⌘F5'),
+    'Maximize pane': ('Ctrl+Alt+Shift+M', u'⌥⇧⌘M'),
+    'Preferences': ('Ctrl+Alt+Shift+P', u'⌥⇧⌘P'),
+    'Quit': ('Ctrl+Q', u'⌘Q'),
+    'Re-run last script': ('F6', u'F6'),
+    'Replace text': ('Ctrl+R', u'⌘R'),
+    'Restart': ('Shift+Alt+R', u'⌥⇧R'),
+    'Run': ('F5', u'F5'),
+    'Save current layout': ('Shift+Alt+S', u'⌥⇧S'),
+    'Show toolbars': ('Alt+Shift+T', u'⌥⇧T'),
+    'Spyder documentation': ('F1', u'F1'),
+    'Switch to breakpoints': ('Ctrl+Shift+B', u'⇧⌘B'),
+    'Switch to editor': ('Ctrl+Shift+E', u'⇧⌘E'),
+    'Switch to explorer': ('Ctrl+Shift+X', u'⇧⌘X'),
+    'Switch to find in files': ('Ctrl+Shift+F', u'⇧⌘F'),
+    'Switch to help': ('Ctrl+Shift+H', u'⇧⌘H'),
+    'Switch to historylog': ('Ctrl+Shift+L', u'⇧⌘L'),
+    'Switch to ipython console': ('Ctrl+Shift+I', u'⇧⌘I'),
+    'Switch to onlinehelp': ('Ctrl+Shift+D', u'⇧⌘D'),
+    'Switch to outline explorer': ('Ctrl+Shift+O', u'⇧⌘O'),
+    'Switch to project explorer': ('Ctrl+Shift+P', u'⇧⌘P'),
+    'Switch to variable explorer': ('Ctrl+Shift+V', u'⇧⌘V'),
+    'Symbol finder': ('Ctrl+Alt+P', u'⌥⌘P'),
+    'Use next layout': ('Shift+Alt+PgDown', u'⌥⇧⇟'),
+    'Use previous layout': ('Shift+Alt+PgUp', u'⌥⇧⇞'),
+    'Enter array inline': ('Ctrl+Alt+M', u'⌥⌘M'),
+    'Enter array table': ('Ctrl+M', u'⌘M'),
+    'Clear line': ('Shift+Escape', u'⇧⎋'),
+    'Clear shell': ('Ctrl+L', u'⌘L'),
+    'Inspect current object': ('Ctrl+I', u'⌘I'),
+    'Blockcomment': ('Ctrl+4', u'⌘4'),
+    'Breakpoint': ('F12', u'F12'),
+    'Close all': ('Ctrl+Shift+W', u'⇧⌘W'),
+    'Close file 1': ('Ctrl+W', u'⌘W'),
+    'Close file 2': ('Ctrl+F4', u'⌘F4'),
+    'Close split panel': ('Alt+Shift+W', u'⌥⇧W'),
+    'Code completion': ('Meta+Space', u'⌃Space'),
+    'Conditional breakpoint': ('Shift+F12', u'⇧F12'),
+    'Copy': ('Ctrl+C', u'⌘C'),
+    'Copy line': ('Shift+Alt+Down', u'⌥⇧↓'),
```

Comment:
You may have to treat Windows and Linux differently for `Copy line` and `Duplicate line`

See https://github.com/spyder-ide/spyder/blob/master/spyder/config/main.py#L373